.github/workflows/test.yml

```diff
@@ -36,14 +36,39 @@ jobs:
       - name: Run Tests
         run: make test
 
+  test-v3:
```

Comment:
Why are we running this in a separate job? Consider including the respective Go versions in the original `test` job. Keep in mind while performing these changes that Go 1.20 is out already.

Reply:
Ditto to the reason I explained in go commands comment.